src/lib/ngc/compile-source-files.ts

```diff
@@ -185,7 +194,10 @@ export async function compileSourceFiles(
   }
 
   if (errorDiagnostics.length) {
+    entryPoint.cache.affectedFiles = new Set(affectedFiles);
```

Comment:
In this case, we should only cache the Angular diagnostics results, we should avoid re-running the diagnostics for files that did not actually change. 

See: https://github.com/angular/angular-cli/blob/11289c4982dfaada860a595fa386d6176f696322/packages/angular/build/src/tools/angular/compilation/aot-compilation.ts#L184

Reply:
I am not really sure how to approach this. How do I know when the content of a template file has changed? The respective *.ts file is not always reported as "affectedFile" when I change the template file and that's why I was forcing it to re-run the diagnostics.

I can't use a cache to store the diagnostics since the diagnostics errors could be outdated.

Reply:
I think I managed to find a simple way to make it work.

Reply:
I think I managed to find a solution for this using the existing `FileCache`.